query/src/main/scala/filodb/query/exec/PeriodicSamplesMapper.scala

```diff
@@ -90,12 +89,11 @@ final case class PeriodicSamplesMapper(start: Long,
   }
 
   // Transform source double or long to double schema
-  override def schema(dataset: Dataset, source: ResultSchema): ResultSchema = {
-
-    if (dataset.options.hasDownsampledData) {
+  override def schema(source: ResultSchema): ResultSchema = {
+    // Special treatment for downsampled gauge schema - return regular timestamp/value
+    if (source.columns == Schemas.dsGauge.dataInfos) {
```

Comment:
Your observation is indeed correct.
This is just a temporary thing.  It will be fixed in the next PR.
It's not possible to fix it without a major refactor, thus I'm delaying it to the next PR, which has the proper fix -- and I'm reluctant to deviate from the plan unless it's a really simple fix, but I can't think of anything simple.

Reply:
Actually I think I can start on some of the intended changes from the plan now, then it won't be broken now.  Watch out for changes.